app/src/main/kotlin/io/homeassistant/companion/android/notifications/PushManager.kt

```diff
@@ -0,0 +1,21 @@
+package io.homeassistant.companion.android.notifications
+
+import io.homeassistant.companion.android.common.notifications.PushProvider
+import javax.inject.Inject
+
+class PushManager @Inject constructor(
+    val providers: Map<String, @JvmSuppressWildcards PushProvider>
+) {
+
+    companion object {
+        private const val FCM_SOURCE = FirebaseCloudMessagingProvider.SOURCE
+    }
+
+    val defaultProvider: PushProvider get() {
+        return providers[FCM_SOURCE]!!
+    }
+
+    suspend fun getToken(): String {
+        return providers[FCM_SOURCE]!!.getToken()
+    }
```

Comment:
Could you already prepare in this PR the fact that FCM is only available in full by keeping this class agnostic from FCM?